e2e/helm.e2e.spec.ts

```diff
@@ -0,0 +1,100 @@
+import path from 'path';
+import {
+  ElectronApplication, BrowserContext, _electron, Page, Locator
+} from 'playwright';
+import { test, expect } from '@playwright/test';
+import { createDefaultSettings, kubectl, helm } from './utils/TestUtils';
+
+let page: Page;
+const defaultReportFolder = path.join(__dirname, 'reports/');
+
+test.describe.serial('Helm Deployment Test', () => {
+  let mainTitle: Locator;
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  const mainTitleSelector = '[data-test="mainTitle"]';
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: path.join(defaultReportFolder, 'pw-trace.zip') });
+    await electronApp.close();
+  });
+
+  test('should load Rancher Desktop App', async() => {
+    mainTitle = page.locator(mainTitleSelector);
+
+    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+  });
+
+  test('should start loading the background services', async() => {
+    const progressBarSelector = page.locator('.progress');
+
+    // Wait until progress bar show up. It takes roughly ~60s to start in CI
+    await progressBarSelector.waitFor({ state: 'visible', timeout: 200_000 });
+    // Wait until progress bar be detached. With that we can make sure the services were started
+    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
+    await expect(progressBarSelector).toBeHidden();
+  });
+
+  test('should check kubernetes API is ready', async() => {
+    const output = await kubectl('cluster-info');
+
+    await expect(output).toMatch(/is running at ./);
+  });
+
+  test('should add helm sample repository', async() => {
+    const helmAddRepoOutput = await helm('repo', 'add', 'bitnami', 'https://charts.bitnami.com/bitnami');
+
+    // Sanity check for local test execution
+    // if the helm repository already exist locally
+    if (helmAddRepoOutput.includes('already exists')) {
+      await expect(helmAddRepoOutput).toContain('"bitnami" already exists with the same configuration, skipping');
+    } else {
+      await expect(helmAddRepoOutput).toContain('"bitnami" has been added to your repositories');
+    }
+  });
+  test('should install helm sample application and check if it was deployed', async() => {
+    const helmInstall = await helm('install', '--wait', '--timeout=20m', 'nginx-sample',
```

Comment:
Can we assume that the chart wasn't previously installed?
Consider either `helm upgrade --install`, or using a scratch namespace.

Reply:
Note that this introduces a dependency on the internet (so if, e.g. bitnami implodes, the tests will fail).

Reply:
Yes, I'm aware about it. I've tried to create a local charts, but it creates a bunch of files (I'm still a newbie on helm btw).
But you're totally right... relying on bitnami+internet could be risky.
I'll try to create a local helm chart (as test asset) and see how it goes. Thanks for the heads up.